test/gtest/uct/cuda/test_cuda_ipc_md.cc

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include <uct/test_md.h>
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+extern "C" {
+#include <uct/cuda/cuda_ipc/cuda_ipc_md.h>
+}
+
+class test_cuda_ipc_md : public test_md {
+protected:
+    static uct_cuda_ipc_rkey_t
+    unpack(uct_md_h md, int64_t uuid0 = 0, int64_t uuid1 = 0)
+    {
+        CUdeviceptr ptr;
+        EXPECT_EQ(CUDA_SUCCESS, cuMemAlloc(&ptr, 64));
+        uct_mem_h memh;
+        EXPECT_UCS_OK(md->ops->mem_reg(md, (void *)ptr, 64, NULL, &memh));
+        uct_cuda_ipc_rkey_t rkey;
+        EXPECT_UCS_OK(md->ops->mkey_pack(md, memh, (void *)ptr, 64, NULL,
+                                         &rkey));
+
+        /* Mock UUID if non-zero value provided */
+        if (uuid0 || uuid1) {
+            int64_t *uuid64 = (int64_t *)rkey.uuid.bytes;
+            uuid64[0]       = uuid0;
+            uuid64[1]       = uuid1;
+        }
+
+        /* cuIpcOpenMemHandle used by cuda_ipc_cache does not allow to open
+         * handle that was created by the same process */
+        EXPECT_EQ(UCS_ERR_UNREACHABLE,
+                  md->component->rkey_unpack(md->component, &rkey, NULL, NULL));
+
+        uct_md_mem_dereg_params_t params;
+        params.field_mask = UCT_MD_MEM_DEREG_FIELD_MEMH;
+        params.memh       = memh;
+        EXPECT_UCS_OK(md->ops->mem_dereg(md, &params));
+
+        EXPECT_EQ(CUDA_SUCCESS, cuMemFree(ptr));
+        return rkey;
+    }
+};
+
+class cuda_context {
+public:
+    cuda_context()
+    {
+        if (cudaSetDevice(0) != cudaSuccess) {
```

Comment:
can it affect the cuda context set in https://github.com/openucx/ucx/blob/6b84982510a89631a215e4fc7b1e19693a89794e/test/gtest/common/mem_buffer.cc#L181